src/app/shared/types/languages.types.ts

```diff
@@ -0,0 +1,3 @@
+import {supportedLanguages} from '../constants/languages.constants';
+
+export type LanguageIdentifier = (typeof supportedLanguages)[number];
```

Comment:
Consistent naming: Either
* rename this `Language` and rename the file to (singular) `language.types.ts`
* rename the file to `language-identifier.types.ts` and change the `supportedLanguages` within `language.constants.ts` to `supportedLanguageIdentifiers` (same with `defaultLanguage`)